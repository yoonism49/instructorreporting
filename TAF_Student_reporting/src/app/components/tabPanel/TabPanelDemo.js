// @flow weak

import React                from 'react';
import TabPanel             from './tabPanel/TabPanel';
import TabPanelHeader       from './tabPanelHeader/TabPanelHeader';
import TabPanelBody         from './tabPanelBody/TabPanelBody';
import TabPanelBodyContent  from './tabPanelBodyContent/TabPanelBodyContent';

const mockHeader = [
  {name: 'Home', tablink: 'home', isActive: true},
  {name: 'About', tablink: 'about', isActive: false},
  {name: 'Profile', tablink: 'profile', isActive: false},
  {name: 'Contact', tablink: 'contact', isActive: false}
];

const TabPanelDemo = () => {
  return (
    <TabPanel>
      <TabPanelHeader tabItems={mockHeader}/>
      <TabPanelBody>
        <TabPanelBodyContent id="home">
          &nbsp;Home
        </TabPanelBodyContent>
        <TabPanelBodyContent id="about">
          &nbsp;About
        </TabPanelBodyContent>
        <TabPanelBodyContent id="profile">
          &nbsp;Profile
        </TabPanelBodyContent>
      </TabPanelBody>
    </TabPanel>
  );
};

export default TabPanelDemo;
