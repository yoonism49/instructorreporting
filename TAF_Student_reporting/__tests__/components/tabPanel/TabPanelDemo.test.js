
'use strict';

import React              from 'react';
import TabPanelDemo       from '../../../src/app/components/tabPanel/TabPanelDemo';
import renderer           from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('TabPanelDemo component', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <TabPanelDemo />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
