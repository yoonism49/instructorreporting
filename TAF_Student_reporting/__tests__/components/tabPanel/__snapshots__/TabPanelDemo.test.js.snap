// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanelDemo component renders as expected 1`] = `
<div>
  <section
    className="panel general"
  >
    <header
      className="panel-heading tab-bg-dark-navy-blue"
    >
      <ul
        className="nav nav-tabs"
      >
        <li
          className="active"
        >
          <a
            aria-expanded="true"
            data-toggle="tab"
            href="#home"
          >
             
            Home
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-expanded="false"
            data-toggle="tab"
            href="#about"
          >
             
            About
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-expanded="false"
            data-toggle="tab"
            href="#profile"
          >
             
            Profile
          </a>
        </li>
        <li
          className=""
        >
          <a
            aria-expanded="false"
            data-toggle="tab"
            href="#contact"
          >
             
            Contact
          </a>
        </li>
      </ul>
    </header>
    <div
      className="panel-body"
    >
      <div
        className="tab-content"
      >
        <div
          className="tab-pane"
          id="home"
        >
           Home
        </div>
        <div
          className="tab-pane"
          id="about"
        >
           About
        </div>
        <div
          className="tab-pane"
          id="profile"
        >
           Profile
        </div>
      </div>
    </div>
  </section>
</div>
`;
